embedded-graphics/src/draw_target.rs

```diff
@@ -0,0 +1,304 @@
+///         // Place an (x, y) pixel at the right index in the framebuffer
+///         let index = coord.x + coord.y * 64;
+///         self.framebuffer[index as usize] = color.luma();
+///
+///         Ok(())
+///     }
+///
+///     fn size(&self) -> Size {
+///         Size::new(64, 64)
+///     }
+/// }
+///
+/// let mut display = ExampleDisplay {
+///     framebuffer: [0; 4096],
+///     iface: SPI1,
+/// };
+///
+/// // Draw a circle centered around `(32, 32)` with a radius of `10` and a white stroke
+/// let circle = egcircle!(
+///     center = (32, 32),
+///     radius = 10,
+///     style = primitive_style!(stroke_color = Gray8::WHITE)
+/// );
+/// circle.draw(&mut display)?;
+///
+/// // Update the display
+/// display.flush().expect("Failed to send data to display");
+/// # Ok::<(), core::convert::Infallible>(())
+/// ```
+///
+/// ## Hardware Acceleration
+///
+/// In addition to defining [`draw_pixel`], an implementation of [`DrawTarget`] can also provide
+/// alternative implementations of drawing methods for graphical primitives. Here is an example of
+/// how a display with accelerated methods can implement [`DrawTarget`]:
+///
+/// ```rust
+/// # use embedded_graphics::prelude::*;
+/// # use embedded_graphics::DrawTarget;
+/// # use embedded_graphics::{egrectangle, primitive_style};
+/// # use embedded_graphics::primitives::rectangle::Rectangle;
+/// # use embedded_graphics::pixelcolor::{Gray8, GrayColor};
+/// # use embedded_graphics::drawable::Pixel;
+/// # use embedded_graphics::style::{PrimitiveStyle, Styled};
+/// #
+/// # struct SPI1;
+/// #
+/// # impl SPI1 {
+/// #     pub fn send_bytes(&self, buf: &[u8]) -> Result<(), ()> {
+/// #         Ok(())
+/// #     }
+/// # }
+/// #
+/// /// A fake display 64px x 64px where each pixel is stored as a single `u8`
+/// struct FastExampleDisplay {
+///     framebuffer: [u8; 64 * 64],
+///     iface: SPI1,
+/// }
+///
+/// impl FastExampleDisplay {
+///     /// Send buffer to the display
+///     pub fn flush(&self) -> Result<(), ()> {
+///         self.iface.send_bytes(&self.framebuffer)
+///     }
+///
+///     /// A HW-accelerated method for drawing rectangles
+///     pub fn fast_rectangle(&self, rect: &Styled<Rectangle, PrimitiveStyle<Gray8>>) {
+///         // Does some speedy drawing
+///     }
+/// }
+///
+/// impl DrawTarget<Gray8> for FastExampleDisplay {
+///     type Error = core::convert::Infallible;
+///
+///     /// Draw a `pixel` that has a color defined as `Gray8`
+///     fn draw_pixel(&mut self, pixel: Pixel<Gray8>) -> Result<(), Self::Error> {
+///         let Pixel(coord, color) = pixel;
+///         // Place an (x, y) pixel at the right index in the framebuffer
+///         let index = coord.x + coord.y * 64;
+///         self.framebuffer[index as usize] = color.luma();
+///
+///         Ok(())
+///     }
+///
+///     fn size(&self) -> Size {
+///         Size::new(64, 64)
+///     }
+///
+///     /// Use the accelerated method when drawing rectangles
```

Comment:
This needs to mention that this overrides the default implementation. IMO this should also be mentioned in the into to this example.